src/h5web/storage-utils.ts

```diff
@@ -0,0 +1,54 @@
+import create, { State, StateCreator, StoreApi, UseStore } from 'zustand';
+
+export interface StorageConfig {
+  key: string;
+  whitelist: string[];
+}
+
+function fromStorage<T>(key: string): T | undefined {
+  try {
+    const value = window.localStorage.getItem(key);
+    return value !== undefined && value !== null
+      ? JSON.parse(value)
+      : undefined;
+  } catch {
+    // Access to local storage may not be authorized or available,
+    // or the data stored may be corrupted.
+    return undefined;
+  }
+}
+
+export function createPersistableState<S extends State>(
+  config: StorageConfig,
+  createState: StateCreator<S>
+): [UseStore<S>, StoreApi<S>] {
+  const { key, whitelist } = config;
+
+  return create<S>((set, get, api) => {
```

Comment:
I decided to wrap zustand's `create` method instead of creating a "classic" middleware. We can change that later if we need more flexibility.